pkg/pillar/zedcloud/lookupproxy.go

```diff
@@ -121,3 +121,56 @@ func LookupProxy(status *types.DeviceNetworkStatus, ifname string,
 	log.Infof("LookupProxy: No proxy configured for port %s", ifname)
 	return nil, nil
 }
+
+// IntfLookupProxyCfg - check if the intf has proxy configured
+func IntfLookupProxyCfg(status *types.DeviceNetworkStatus, ifname, downloadURL string) string {
+	var mayChange bool
+	// if proxy is not on the intf, then don't change anything
+	// if download URL has "http://" or "https://" then no change here regardless of proxy
+	// if there is proxy on this intf, treeat empty url scheme as for https,
+	// treat passed in download url schem docker://" and maybe sftp://just as https
+	if strings.Contains(downloadURL, "http") {
```

Comment:
I completely missed that this matches htttp and https
Would be clearer (and avoid false matches on foo.bar/http-is-a-bad-protocol.html) to
1) compare http:// and https://
2) do a HasPrefix instead of Contains.

#3 applies to the docker comparison as well

Reply:
Hmm since you are parsing the URL below, why not do it here and then a case statement on the Scheme from the parsed URL?

Reply:
ok, will change the string compare. this code is saying, if the 'download URL' already is 'http' or 'https', this function does not need to modify anything, the LookupProxy() will handle that properly.